### ISSUE TITLE
Trend visualization comparison not working on multi-aggregated date

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#13284]
**Describe the bug**
When summarizing a date column more than once, then Trend visualization doesn't understand it's a date anymore and stops showing comparison to previous period.

**To Reproduce**
1. Custom question > Sample Dataset > Orders
2. Summarize "Max of:Total" by "CreatedAt:Month"
3. Summarize "Sum of:Max of Total" by "CreatedAt"
![image](https://user-images.githubusercontent.com/1447303/93789587-556d1d00-fc32-11ea-9925-345d27035f2c.png)
4. Visualize as Trend - the comparison says `Nothing to compare for the previous day.`
![image](https://user-images.githubusercontent.com/1447303/93789707-7a619000-fc32-11ea-952a-5853def36d32.png)
5. Even though it's returning the month with date formatting:
![image](https://user-images.githubusercontent.com/1447303/93789818-9ebd6c80-fc32-11ea-83b6-19a98bf698d0.png)

**Expected behavior**
Should behave just like when there's just a single Summarize.

**Information about your Metabase Installation:**
Metabase 0.36.6

**Additional context**
https://discourse.metabase.com/t/trend-graphic-with-double-summarize/11675

**Comment by markbastian** [comment]
I am unable to reproduce this. However, in the second stage of the query, the "Created At" field _does_ retain the fact that it is a temporal value and the ability to group by time unit.

<img width="1157" alt="image" src="https://github.com/metabase/metabase/assets/8381441/4f588119-efdf-4c9a-914d-0d8c6326defe">

This makes me wonder if this is fixed. The trend also works:

<img width="795" alt="image" src="https://github.com/metabase/metabase/assets/8381441/3783469b-03bb-4c2d-ac66-92d8f533ecae">

Note that the aggregation on the trend delta must be at last as large as the previous stage or it will not show anything to compute. For example, grouping by month then by day doesn't make sense, but grouping by month then year might.

**Comment by markbastian** [comment]
Note: I also set the semantic type of the Created At to no semantic type and everything still works as above.

**Comment by Tony-metabase** [comment]
Yep the second stage is already different and offers time granularity ... Closing with unable to replicate
